src/bin/pg_dump/dump_babel_utils.c

```diff
@@ -757,11 +873,18 @@ prepareForBabelfishDatabaseDump(Archive *fout, SimpleStringList *schema_include_
 	}
 
 	query = createPQExpBuffer();
-	/* Get oids of all the Babelfish catalog configuration tables */
+	/*
+	 * Get oids of all the Babelfish catalog configuration tables.
+	 * See comments for updateExtConfigArray above for more details
+	 * about why we are excluding/including certain tables in the query
+	 * below.
+	 */
 	appendPQExpBufferStr(query,
-						 "SELECT unnest(extconfig)::oid "
-						 "FROM pg_catalog.pg_extension "
-						 "WHERE extname = 'babelfishpg_tsql';");
+						 "WITH tableoids AS ("
+						 "SELECT unnest(extconfig)::oid AS id "
+						 "FROM pg_catalog.pg_extension WHERE extname = 'babelfishpg_tsql') "
+						 "SELECT id FROM tableoids WHERE id != 'sys.babelfish_configurations'::regclass " /* Exclude babelfish_configurations table */
+						 "UNION SELECT 'sys.babelfish_authid_user_ext'::regclass AS id "); /* Include babelfish_authid_user_ext table */
```

Comment:
The above update is to improve this logic now although it will only be relevant when we support cross-version dump/restore. Currently, this has been added to allow TAP tests to run successfully.